### PR TITLE
Update the Access behaviour to support slices

### PR DESCRIPTION
**Pull request by kipcole9**
The PR updates the Access behaviour for images to allow slices in a manner similar to Nx. See the documentation in the `Vix.Vips.Image` module and the tests in the `access_test.exs` files.

### CODE REVIEW

**Comment by kipcole9** [comment]
In the current implementation, errors in the Access protocol return `nil`. In `Nx` they raise and exception.  Perhaps Vix should also raise?  What do you think?

**Comment by kipcole9** [comment]
The failing tests are for ranges like `1..10//1` which aren't supported on older releases. I'll have a think about how to work around that and push another commit when I work it out.

**Comment by kipcole9** [comment]
Phew, finally testing green in CI. Hardest part was working around syntax errors on older Elixir releases. Anyway, ready for your review and comment now.

**Comment by akash-akya** [comment]
This looks fantastic @kipcole9, will take a closer look tomorrow and merge.
Thank you!

**Comment by akash-akya** [comment]
I think we should too raise an exception similar to `Nx`, just for the sake of consistency.

File: lib/vix/vips/image.ex [hunk 1]
```diff
@@ -83,6 +144,42 @@ defmodule Vix.Vips.Image do
     end
   end
 
+  def fetch(image, %Range{first: first, last: last} = range) when first >= 0 and last < 0 do
```
Comment (akash-akya) [inline lib/vix/vips/image.ex:147]: We should support `-2..-1` ?

Comment (kipcole9) [reply lib/vix/vips/image.ex:147]: Good catch, will add that.

**Comment by kipcole9** [comment]
> I think we should too raise an exception similar to `Nx`, just for the sake of consistency.

Will update to raise, I agree its good to be consistent.

**Comment by kipcole9** [comment]
I'm still working on it. After review I'm not very happy with parts of the code. I will revise and update over the weekend - including raising on error.

**Comment by kipcole9** [comment]
I have finally updated the Access protocol implementation which I think reads much more clearly now. And the implementation now raises on error rather than returning `nil`. This means `Vix` is more consistent with `Nx` and `eVision`.

**Comment by kipcole9** [comment]
And now all checks are passing so subject to your review and and resulting updates I think this is good to go. Sorry for leaving this open for so long.

**Comment by akash-akya** [comment]
Merging this. We can address the pending things separately.
Thanks!

**Comment by kipcole9** [comment]
@akash-akya, I will work today on a separate PR on the changes you suggested.

**Comment by akash-akya** [comment]
I am already working on it. I just wanted a secondary opinion on this :)

**Comment by kipcole9** [comment]
Ah, ok, understood. I read the updated Nx docs and your proposal makes perfect sense and much better than an `:all` specifier.

**Comment by akash-akya** [comment]
Changes are here: https://github.com/akash-akya/vix/pull/81
Going to merge this, please let me know if you have any concerns.

**Pull request merged**